src/window.c

```diff
@@ -59,6 +76,20 @@ void window_draw_char(struct window *w, int x, int y, char ch, struct graphics_c
     window_end_draw();
 }
 
+void window_draw_string(struct window *w, int x, int y, char *str, struct graphics_color fgcolor,
+                        struct graphics_color bgcolor) {
+    int pos_h = 0;
+    int pos_w = 0;
+    while (*str) {
+        window_draw_char(w, x + pos_w, y + pos_h, *str, fgcolor, bgcolor);
+        pos_w += CHARACTER_W;
+        if (*str == '\n') {
+            pos_h += CHARACTER_H;
```

Comment:
should we reset pos_w = 0 here? right now it seems like a newline just drops you down without resetting your x position for drawing the next character.